### PR TITLE
Convert NodeDescriptor to a Proto.

### PR DESCRIPTION
**Pull request by BramGruneir**
I'd like to pull out RawAddr, but that would be too big and too involved for this PR.

### CODE REVIEW

File: kv/dist_sender_test.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ var testRangeDescriptor = proto.RangeDescriptor{
 	},
 }
 
-var testAddress = util.MakeRawAddr("tcp", "node1:8080")
+var testAddress, _ = net.ResolveTCPAddr("tcp", "localhost:8080")
```
Comment (tschottdorf) [inline kv/dist_sender_test.go]: why this change? I'm assuming it wasn't actually used to connect to.

File: kv/dist_sender_test.go [hunk 2]
```diff
@@ -209,11 +212,17 @@ func TestSendRPCOrder(t *testing.T) {
 			Replicas: nil,
 		}
 		for i := int32(1); i <= 5; i++ {
-			addr := util.MakeRawAddr("tcp", fmt.Sprintf("%d", i))
+			addr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("localhost:%d", i))
```
Comment (tschottdorf) [inline kv/dist_sender_test.go]: I'd prefer using bogus names to make it clear nobody is actually connecting there. Something like `fmt.Sprintf("node%d", i)`.

**Comment by tschottdorf** [comment]
LGTM

File: kv/dist_sender_test.go (hunk 2 above)
Comment (BramGruneir) [inline kv/dist_sender_test.go]: I'm going to change this back.  The problem is the way in which go handles addresses.  You can't have an unresolved net.addr, which is why we have the RawNetAddr.  But the documentation in RawNetAddr was not clear.  I'm going to send a PR later today which will change the name to UnresolvedNetAddr that will make all of this a lot clearer.

File: kv/dist_sender_test.go (hunk 1 above)
Comment (BramGruneir) [inline kv/dist_sender_test.go]: I'm going to revert this as well.
